csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -677,7 +550,9 @@ public void RefreshPublishedEndpoints()
 
             try
             {
-                UpdateLocatorRegistry(_locatorInfo, CreateDirectProxy(new Identity("dummy", ""), IObjectPrx.Factory));
+                _ = UpdateLocatorRegistryAsync(_locatorInfo,
```

Comment:
We shouldn't ignore the returned task here and wait on it instead otherwise the exception handling isn't correct. Also we should fix ComputePublishedEndpoints to be Async (RouterInfo.GetServerEndpoints needs fixing as well to be async).

Reply:
Fixed